Fix: Ensure proper UTF-8 handling for renderer name

VergilGao opened the pull request:
When setting the renderer name to non-ASCII characters (such as Chinese),  gmediarender failed due to invalid byte sequence conversion. The issue was  identified in https://github.com/hzeller/gmrender-resurrect/issues/81, where  it was suggested that explicitly setting the locale in the main function  could resolve the problem.

This patch adds:
- `#include <locale.h>` to ensure locale handling is available.
- `setlocale(LC_ALL, "")` at the beginning of `main()` to use the system locale.

With this change, gmediarender correctly processes UTF-8 encoded renderer  names, preventing command line parsing errors.

Reference: https://github.com/hzeller/gmrender-resurrect/issues/81

hzeller commented on src/main.c:
```diff
@@ -156,6 +157,7 @@ static void do_show_version(void)
 
 static gboolean process_cmdline(int argc, char **argv)
 {
+	setlocale(LC_ALL, "");
```
This certainly requires a comment of sorts describing why this is necessary.
Is this documented in https://docs.gtk.org/glib/ somewhere ?

VergilGao replied:
i found some docs here:
https://docs.gtk.org/glib/i18n.html

VergilGao replied:
@coldtobi  It makes sense; calling setlocale() directly in main() provides a clearer and more straightforward implementation.

VergilGao committed:
Fix: Ensure proper UTF-8 handling for renderer name

When setting the renderer name to non-ASCII characters (such as Chinese),
gmediarender failed due to invalid byte sequence conversion. The issue was
identified in https://github.com/hzeller/gmrender-resurrect/issues/81, where
it was suggested that explicitly setting the locale in the main function
could resolve the problem.

This patch adds:
- `#include <locale.h>` to ensure locale handling is available.
- `setlocale(LC_ALL, "")` at the beginning of `main()` to use the system locale.

With this change, gmediarender correctly processes UTF-8 encoded renderer
names, preventing command line parsing errors.

Reference: https://github.com/hzeller/gmrender-resurrect/issues/81

hzeller reviewed (approved):
Thanks
